src/lib/swapper/swappers/OneInchSwapper/utils/types.ts

```diff
@@ -0,0 +1,75 @@
+import type { EvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { Trade } from 'lib/swapper/api'
+
+export type OneInchQuoteApiInput = {
+  fromTokenAddress: string
+  toTokenAddress: string
+  amount: string
+}
+
+export type OneInchSwapApiInput = OneInchQuoteApiInput & {
+  slippage: number
+  fromAddress: string
+  referrerAddress: string
+  allowPartialFill: boolean
+  disableEstimate: boolean
```

Comment:
https://docs.1inch.io/docs/aggregation-protocol/api/swap-params <- swap params take in a `fee` param that can be used for our opt-out donations.

Reply:
> this percentage of fromTokenAddress token amount will be sent to referrerAddress, the rest will be used as input for a swap
min: 0; max: 3; default: 0
example to set a fee to 1.5%: &fee=1.5